tools/release/adot-operator-images-mirror/docker_auth.go

```diff
@@ -1,22 +1,28 @@
 package main
 
 import (
-	"fmt"
-	"log"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"strings"
 
-	docker "github.com/fsouza/go-dockerclient"
+	"github.com/docker/docker/api/types"
 )
 
-func getDockerCredentials(registry string) (*docker.AuthConfiguration, error) {
-	authOptions, err := docker.NewAuthConfigurationsFromDockerCfg()
+func getDockerCredentials(authToken string) (string, error) {
+	decoder := base64.NewDecoder(base64.StdEncoding, strings.NewReader(authToken))
+	credentials, err := io.ReadAll(decoder)
 	if err != nil {
-		log.Fatal(err)
+		return "", err
 	}
-
-	creds, ok := authOptions.Configs[registry]
-	if !ok {
-		return nil, fmt.Errorf("no auth found for %s", registry)
+	parts := strings.Split(string(credentials), ":")
+	authConfig := types.AuthConfig{
+		Username: parts[0],
+		Password: parts[1],
```

Comment:
Do we know for sure that `parts[0]` and `parts[1]` will defined?

Reply:
Yes, based on the [ECR documentation](https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_AuthorizationData.html):
> When the string is decoded, it is presented in the format user:password for private registry authentication using docker login.

Reply:
It is possible that this function was provided a b64-encoded string that is not an auth token and does not contain ':'.  The `strings.Split()` should probably be `strings.Splitn(creds, 2)` to ensure no more than two parts (i.e., if the password contains ':' do not split it into three pieces) and then there should be an assertion that `len(parts) == 2`.

Reply:
I've added the check in.